packages/brain/src/modules/cron/cron.ts

```diff
@@ -0,0 +1,34 @@
+import { setInterval, clearInterval } from "timers";
+import logger from "../logger/index.js";
+
+export class CronJob {
+  private task: NodeJS.Timeout | null = null;
+  private interval: number;
+
+  constructor(interval: number, private jobFunction: () => Promise<void>) {
+    this.interval = interval;
```

Comment:
what if interval is negative or 0? will the cronjob execute nonstop? Id put something to prevent it like
```
if (interval <= 0) {
      throw new Error("Interval must be a positive number.");
    }
```